### PR TITLE
feat: adds secrets-store-sync-controller project

### PR DESCRIPTION
**Pull request by nilekhc**
- `secrets-store-sync-controller` is the new spin-off project from `secrets-store-csi-driver` sponsored by SigAuth and added to kubernetes-sigs
- onboarding the project, so we can start publishing the images to `registry.k8s.io`
- part of: https://github.com/kubernetes-sigs/secrets-store-sync-controller/issues/5

/cc @aramase @enj 
/assign @dims

### CODE REVIEW

**Comment by nilekhc** [comment]
@upodroid Could you PTAL when you get a chance? Thanks.

File: registry.k8s.io/manifests/k8s-staging-secrets-store-sync/promoter-manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# google group for gcr.io/k8s-staging-secrets-store-sync is k8s-infra-staging-secrets-store-sync@kubernetes.io
+registries:
+- name: gcr.io/k8s-staging-secrets-store-sync
```
Comment (upodroid) [inline registry.k8s.io/manifests/k8s-staging-secrets-store-sync/promoter-manifest.yaml:3]:
```suggestion
- name: us-central1-docker.pkg.dev/k8s-staging-images/secrets-store-sync
```

Comment (upodroid) [reply registry.k8s.io/manifests/k8s-staging-secrets-store-sync/promoter-manifest.yaml:3]: Please push your staging images to this registry please.

**Review by upodroid — CHANGES_REQUESTED** [review]

**Comment by nilekhc** [comment]
@upodroid I have updated the PR. PTAL. Thanks.

File: registry.k8s.io/images/k8s-staging-secrets-store-sync/generate.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly repo="gcr.io/k8s-staging-secrets-store-sync"
+# release candidates will be created for >=1.0.0 releases that we want published to the prod registry
+readonly tag_filter="(tags~^v[0-9]+.[0-9]+.[0-9]+$) OR (tags~^v[0-9]+.[0-9]+.[0-9]+-rc.[0-9]+$ AND tags >= v1.0.0)"
+readonly images=(
+    secrets-store-sync-controller
```
Comment (upodroid) [inline registry.k8s.io/images/k8s-staging-secrets-store-sync/generate.sh:25]:
```suggestion
    controller
```

File: registry.k8s.io/images/k8s-staging-secrets-store-sync/generate.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly repo="gcr.io/k8s-staging-secrets-store-sync"
```
Comment (upodroid) [inline registry.k8s.io/images/k8s-staging-secrets-store-sync/generate.sh:21]: us-central1-docker.pkg.dev/k8s-staging-images/secrets-store-sync
```suggestion
readonly repo="gcr.io/k8s-staging-secrets-store-sync"
```

**Review by upodroid — APPROVED** [review]
Thanks

**Pull request merged**